generators/generator-base-private.js

```diff
@@ -795,55 +795,47 @@ module.exports = class extends Generator {
      * @return {string} version - retrieved version or empty string if not found
      */
     findBlueprintVersion(blueprintPkgName) {
-        let packageJsonPath = path.join(process.cwd(), 'node_modules', blueprintPkgName, 'package.json');
-        try {
-            if (!fs.existsSync(packageJsonPath)) {
-                this.debug('using global module as local version could not be found in node_modules');
-                packageJsonPath = path.join(blueprintPkgName, 'package.json');
+        const blueprintGeneratorName = blueprintPkgName.replace('generator-', '');
+        // Find any subGenerator that belongs to the blueprint
+        const subGenerator = Object.values(this.env.getGeneratorsMeta()).find(subgen =>
+            subgen.namespace.includes(`${blueprintGeneratorName}:`)
+        );
+
+        if (subGenerator && subGenerator.resolved) {
+            // Try to retrieve package.json from subGenerator resolved path <subGenerator path>:
+            // <subGenerator path>/../package.json for pattern "<package root>/index.js"
+            // <subGenerator path>/../../package.json for pattern "<package root>/generators/<subGenerator path>/index.js"
+            // <subGenerator path>/../../../package.json for pattern "<package root>/lib/generators/<subGenerator path>/index.js"
+            // eslint-disable-next-line no-restricted-syntax
+            for (const i of _.range(1, 4)) {
+                const packageJsonPath = path.join(subGenerator.resolved, '../'.repeat(i), 'package.json');
+                if (fs.existsSync(packageJsonPath)) {
+                    // eslint-disable-next-line global-require,import/no-dynamic-require
+                    const blueprintPackagejs = require(packageJsonPath);
+                    return blueprintPackagejs.version;
+                }
             }
-            // eslint-disable-next-line global-require,import/no-dynamic-require
-            const packagejs = require(packageJsonPath);
-            return packagejs.version;
-        } catch (err) {
-            this.debug('ERROR:', err);
-            this.warning(`Could not retrieve version of blueprint '${blueprintPkgName}'`);
-            return '';
         }
```

Comment:
With yeoman-environment v2.6.0:
const packageJsonPath = path.join(this.env. getPackagePath(blueprintGeneratorName), 'package.json');
 if (fs.existsSync(packageJsonPath)) {
    // eslint-disable-next-line global-require,import/no-dynamic-require
    const blueprintPackagejs = require(packageJsonPath);
    return blueprintPackagejs.version;
}